### ISSUE TITLE
[TF2]  broken half lambert lighting since launch

### ISSUE THREAD

**Issue by condor00fr** [ValveSoftware/Source-1-Games#4471]
This comes from a shader mod from Gamebanana:
https://gamebanana.com/mods/36761

and Here's a quote from the moder itself:

"Just to clear up any miscommunication with the name of this mod, this lighting is pretty much fixed half lambert lighting from prerelease tf2, as Valve broke the half lambert lighting on launch. Essentially this is what TF2's lighting is intended  to look like, but valve never bothered to fix it post release. This mod also works perfectly fine on sv_pure 1 servers since it doesn't really change assets from the game."

**Comment by wurlmon** [comment]
I don't think this is how TF2 is supposed to look like, considering half lamber is an optional setting when creating cosmetics.
TF2 uses the current setup (phong) for a reason, and it probably is that phong looks nicer for most things

**Comment by Whurrhurr** [comment]
> I don't think this is how TF2 is supposed to look like, considering half lamber is an optional setting when creating cosmetics. TF2 uses the current setup (phong) for a reason, and it probably is that phong looks nicer for most things

$halflambert 1 isn't optional, excluding it from your cosmetic is the same thing as including it and setting it to 0. it's only noticeable when you disable phong, the mat_phong 0 $halflambert fix mod on game banana's description of this parameter is: "This parameter is supposed to disable Half-Lambertian lighting calculations, used to create a smooth value around models from back to front, rather than just in the direction of the light. However, a change in 2007 made it so that this parameter is ignored when $phong is enabled, so regardless of what the VMT says, it treats it as if $halflambert is set to 1." most of tf2's lightwarps (at least the ones it had at launch anyway) are supposed to mimic lambertian shading while giving greater control on stylistic shading. and some of the lightwarps we have now look **hideous** with phong disabled.
in case you want to see the mod in question and some before/after pictures, here it is: https://gamebanana.com/mods/345643

also being able to exclude something that messes up lightwarp shading if excluded/set to 0 instead of 1 feels like a mistake.

**Comment by Wosheereal** [comment]
Are you really trying to act smart here towards someone who knows what they are talking about? This is not the intended lighting of the game, even in the Illustrative Rendering video that was published in 2007 shows the lighting looking like how it is now.
source: https://youtu.be/ebvbVO1QE5o?t=127

**Comment by condor00fr** [comment]
.....

A guy can be wrong you know....... no need to be mean

**Comment by BreavyTF2** [comment]
I'd say a fair compromise would be letting "$halflambert" "0" work with phong params.
